python/nmodl/ode.py

```diff
@@ -344,6 +344,55 @@ def solve_lin_system(
     return code, new_local_vars
 
 
+def finite_difference_step_variable(sym):
+    return f"{sym}_delta_"
+
+
+def discretize_derivative(expr):
+    if isinstance(expr, sp.Derivative):
+        x = expr.args[1][0]
+        dx = sp.symbols(finite_difference_step_variable(x))
+        return expr.as_finite_difference(dx)
+    else:
+        return expr
+
+
+def transform_expression(expr, transform):
+    if expr.args is tuple():
```

Comment:
```suggestion
    if not expr.args:
```
An empty tuple is falsy, and using `is` for stuff other than singletons (like `None`) can be error-prone.

Reply:
Spec promises it's an empty something. Way too many things are falsey, including `0` to make that mistake.